components/config-mgmt-service/backend/elastic/runs.go

```diff
@@ -71,12 +72,142 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
+	}
+	copiedFilters["exists"] = []string{"false"}
+	mainQuery := newBoolQueryFromFilters(copiedFilters)
+
+	deletedNodePeroids := make([]backend.CountPeroid, getNumberOf24hBetween(startTime, endTime))
+
+	dateRangeAgg := elastic.NewDateRangeAggregation().Field(backend.Timestamp).
+		Format("yyyy-MM-dd'T'HH:mm:ssZ")
+
+	for index := 0; index < len(deletedNodePeroids); index++ {
+		end := startTime.Add(time.Hour * 24 *
+			time.Duration(index)).Add(time.Hour * 24).Add(-time.Millisecond)
+
+		dateRangeAgg.AddUnboundedFromWithKey(strconv.Itoa(index), end.Format(time.RFC3339))
+	}
+
+	searchResult, err := es.client.Search().
+		Index(IndexNodeState).
+		Query(mainQuery).
+		Aggregation(aggTag, dateRangeAgg).
+		Do(context.Background())
+	if err != nil {
+		return []backend.CountPeroid{}, err
+	}
+
+	rangeAggRes, found := searchResult.Aggregations.Range(aggTag)
+	if !found {
+		// This case is if there are no runs for the entire range of the time series
+		// We are creating the buckets manually with zero check-ins
+		for index := 0; index < len(deletedNodePeroids); index++ {
+			deletedNodePeroids[index].Start = startTime.Add(time.Hour * 24 * time.Duration(index))
+			deletedNodePeroids[index].End = startTime.Add(time.Hour * 24 *
+				time.Duration(index)).Add(time.Hour * 24).Add(-time.Millisecond)
+			deletedNodePeroids[index].Count = 0
+		}
+		return deletedNodePeroids, nil
+	}
+
+	if len(rangeAggRes.Buckets) != len(deletedNodePeroids) {
+		return []backend.CountPeroid{}, errors.NewBackendError(
+			"The number of buckets found is incorrect expected %d actual %d",
+			len(deletedNodePeroids), len(rangeAggRes.Buckets))
+	}
+
+	for index, bucket := range rangeAggRes.Buckets {
+		deletedNodePeroids[index].Count = int(bucket.DocCount)
+		deletedNodePeroids[index].Start = startTime.Add(time.Hour * 24 * time.Duration(index))
+		deletedNodePeroids[index].End = startTime.Add(time.Hour * 24 *
+			time.Duration(index)).Add(time.Hour * 24).Add(-time.Millisecond)
+	}
+
+	return deletedNodePeroids, nil
+}
+
+// GetCreateCountsTimeSeries - creates a daily time series of the total amount of nodes that have
+// been created within or before the time range. The time series is between the startTime and endTime provided.
+// The startTime and endTime must atleast be 24 hours apart. When greater than 24 hours, it must be in
+// multiples 24 hour blocks.
+func (es Backend) GetCreateCountsTimeSeries(startTime, endTime time.Time,
+	filters map[string][]string) ([]backend.CountPeroid, error) {
+	var (
+		aggTag    = "date_range"
+		mainQuery = newBoolQueryFromFilters(filters)
+	)
+
+	createNodePeroids := make([]backend.CountPeroid, getNumberOf24hBetween(startTime, endTime))
+
+	// This is using the custom_search_aggs_bucket_date_range because it need to set the missing option
+	dateRangeAgg := NewDateRangeAggregation().Field(backend.Created).
+		Format("yyyy-MM-dd'T'HH:mm:ssZ").
+		Missing(time.Time{}.Format(time.RFC3339)) // count all nodes that do not have a create date
```

Comment:
The "missing" option is what was not implemented in the olivere code. This is the reason the olivere code was copied.